google/data_source_storage_object_signed_url.go

```diff
@@ -68,7 +69,7 @@ func dataSourceGoogleSignedUrl() *schema.Resource {
 				Type:         schema.TypeString,
 				Optional:     true,
 				Default:      "GET",
-				ValidateFunc: validateHttpMethod,
+				ValidateFunc: validation.StringInSlice([]string{"GET", "HEAD", "PUT", "DELETE"}, true),
```

Comment:
Given that not all of the http verbs are here, I don't think we should add a custom validator yet;  if there are several other validators with identical parameters in the future, that is something we should consider, though.